src/main/java/edu/illinois/library/cantaloupe/processor/PdfBoxProcessor.java

```diff
@@ -187,6 +186,25 @@ public void put(COSObject indirect, PDXObject xobject) {
         }
     }
 
+    private MemoryUsageSetting getMemoryUsageSetting() {
+        final Configuration config = Configuration.getInstance();
+        final boolean scratchFileEnabled =
+                config.getBoolean(Key.PROCESSOR_PDF_SCRATCH_FILE_ENABLED, false);
+        final long maxMainMemoryBytes =
+                config.getLong(Key.PROCESSOR_PDF_MAX_MEMORY_BYTES, -1);
+        final long maxStorageBytes =
+                config.getLong(Key.PROCESSOR_PDF_MAX_STORAGE_BYTES, -1);
+        final String scratchFileLocation =
+                config.getString(Key.PROCESSOR_PDF_SCRATCH_LOCATION);
+        if (scratchFileEnabled) {
+            File filePath = new File(scratchFileLocation);
+            return MemoryUsageSetting.setupMixed(maxMainMemoryBytes,
+                    maxStorageBytes).setTempDir(filePath);
+        } else {
+            return MemoryUsageSetting.setupMainMemoryOnly(maxMainMemoryBytes);
```

Comment:
I noticed some unexpected behavior here. When I set `processor.pdf.scratch_file_enabled = false` and `processor.pdf.max_memory_bytes` to a small value, it may result in an incompletely rendered PDF. For example, I have a PDF in which all the text renders but the main image ends up blank. So I think it would be best not to constrain the in-memory size when scratch files are disabled.

Reply:
Will fix.